internal/dag/builder_test.go

```diff
@@ -6704,6 +6716,32 @@ func TestValidateHeaderAlteration(t *testing.T) {
 	}
 }
 
+func TestBuilderRunsProcessorsInOrder(t *testing.T) {
+	var got []string
+
+	b := Builder{
+		Processors: []Processor{
+			&pluggableProcessor{runFunc: func(_ *Builder) { got = append(got, "foo") }},
+			&pluggableProcessor{runFunc: func(_ *Builder) { got = append(got, "bar") }},
+			&pluggableProcessor{runFunc: func(_ *Builder) { got = append(got, "baz") }},
+			&pluggableProcessor{runFunc: func(_ *Builder) { got = append(got, "abc") }},
+			&pluggableProcessor{runFunc: func(_ *Builder) { got = append(got, "def") }},
+		},
+	}
+
+	b.Build()
+
+	assert.Equal(t, []string{"foo", "bar", "baz", "abc", "def"}, got)
+}
+
+type pluggableProcessor struct {
+	runFunc func(builder *Builder)
+}
+
+func (p *pluggableProcessor) Run(builder *Builder) {
+	p.runFunc(builder)
+}
```

Comment:
I've generally found it's pretty useful to add a function adaptor at the API layer, eg.
```Go
type BuildProcessorFunc func(*Builder)

func (b BuildProcessorFunc) Build(builder *Builder) {
    if b != nil {
        b(builder)
    }
}
```

Maybe add this in a subsequent PR?